Makefile

```diff
@@ -0,0 +1,156 @@
+	$(info Done!)
+
+$(BUILDDIR)/$(BINARY_NAME): $(BUILDDIR)
+	@cd $(BASE)/cmd/$(BINARY_NAME) && $(GO) build -o $(BUILDDIR)/$(BINARY_NAME) -v
+
+
+# Tools
+
+GOLINT = $(GOBIN)/golint
+$(GOBIN)/golint: | $(BASE) ; $(info  building golint...)
+	$Q go get -u golang.org/x/lint/golint
+
+GOCOVMERGE = $(GOBIN)/gocovmerge
+$(GOBIN)/gocovmerge: | $(BASE) ; $(info  building gocovmerge...)
+	$Q go get github.com/wadey/gocovmerge
+
+GOCOV = $(GOBIN)/gocov
+$(GOBIN)/gocov: | $(BASE) ; $(info  building gocov...)
+	$Q go get github.com/axw/gocov/...
+
+GOCOVXML = $(GOBIN)/gocov-xml
+$(GOBIN)/gocov-xml: | $(BASE) ; $(info  building gocov-xml...)
+	$Q go get github.com/AlekSi/gocov-xml
+
+GO2XUNIT = $(GOBIN)/go2xunit
+$(GOBIN)/go2xunit: | $(BASE) ; $(info  building go2xunit...)
+	$Q go get github.com/tebeka/go2xunit
+
+
+# Tests
+
+TEST_TARGETS := test-default test-bench test-short test-verbose test-race
+.PHONY: $(TEST_TARGETS) test-xml check test tests
+test-bench:   ARGS=-run=__absolutelynothing__ -bench=. ## Run benchmarks
+test-short:   ARGS=-short        ## Run only short tests
+test-verbose: ARGS=-v            ## Run tests in verbose mode with coverage reporting
+test-race:    ARGS=-race         ## Run tests with race detector
+$(TEST_TARGETS): NAME=$(MAKECMDGOALS:test-%=%)
+$(TEST_TARGETS): test
+check test tests: fmt lint vendor | $(BASE) ; $(info  running $(NAME:%=% )tests...) @ ## Run tests
+	$Q cd $(BASE) && $(GO) test -timeout $(TIMEOUT)s $(ARGS) $(TESTPKGS)
+
+test-xml: fmt lint vendor | $(BASE) $(GO2XUNIT) ; $(info  running $(NAME:%=% )tests...) @ ## Run tests with xUnit output
+	$Q cd $(BASE) && 2>&1 $(GO) test -timeout 20s -v $(TESTPKGS) | tee test/tests.output
+	$(GO2XUNIT) -fail -input test/tests.output -output test/tests.xml
+
+COVERAGE_MODE = atomic
+COVERAGE_PROFILE = $(COVERAGE_DIR)/profile.out
+COVERAGE_XML = $(COVERAGE_DIR)/coverage.xml
+COVERAGE_HTML = $(COVERAGE_DIR)/index.html
+.PHONY: test-coverage test-coverage-tools
+test-coverage-tools: | $(GOCOVMERGE) $(GOCOV) $(GOCOVXML)
+test-coverage: COVERAGE_DIR := $(CURDIR)/test/coverage.$(shell date -u +"%Y-%m-%dT%H:%M:%SZ")
+test-coverage: fmt lint vendor test-coverage-tools | $(BASE) ; $(info  running coverage tests...) @ ## Run coverage tests
+	$Q mkdir -p $(COVERAGE_DIR)/coverage
+	$Q cd $(BASE) && for pkg in $(TESTPKGS); do \
+		$(GO) test \
+			-coverpkg=$$($(GO) list -f '{{ join .Deps "\n" }}' $$pkg | \
+					grep '^$(PACKAGE)/' | grep -v '^$(PACKAGE)/vendor/' | \
+					tr '\n' ',')$$pkg \
+			-covermode=$(COVERAGE_MODE) \
+			-coverprofile="$(COVERAGE_DIR)/coverage/`echo $$pkg | tr "/" "-"`.cover" $$pkg ;\
+	 done
+	$Q $(GOCOVMERGE) $(COVERAGE_DIR)/coverage/*.cover > $(COVERAGE_PROFILE)
+	$Q $(GO) tool cover -html=$(COVERAGE_PROFILE) -o $(COVERAGE_HTML)
+	$Q $(GOCOV) convert $(COVERAGE_PROFILE) | $(GOCOVXML) > $(COVERAGE_XML)
+
+.PHONY: lint
+lint: vendor | $(BASE) $(GOLINT) ; $(info  running golint...) @ ## Run golint
```

Comment:
'running golint' complains error:
/bin/sh:line1: $(HOME)/sriov-network-device-plugin/bin/golint: No such file or directory.

Reply:
Can you run make clean then make?
golint supposed be installed if not found. Please update me on this. Thanks

Reply:
Tried with make clean;make, still failed.
Seems it's looking for 'golint' under root dir of sriov-network-device-plugin repo : /root/sriov-network-device-plugin/bin/golint, but there is no bin/ dir and golint doesn't exist.

[root@test sriov-network-device-plugin]# make
running gofmt...
setting GOPATH...
building golint...
running golint...
/bin/sh:line1: /root/sriov-network-device-plugin/bin/golint: No such file or directory
github.com/intel/sriov-network-device-plugin/vendor/golang.org/x/net/context
github.com/intel/sriov-network-device-plugin/vendor/github.com/golang/protobuf/proto
github.com/intel/sriov-network-device-plugin/vendor/github.com/golang/glog
github.com/intel/sriov-network-device-plugin/vendor/golang.org/x/net/http2/hpack
github.com/intel/sriov-network-device-plugin/vendor/golang.org/x/text/transform
github.com/intel/sriov-network-device-plugin/vendor/golang.org/x/text/unicode/bidi
github.com/intel/sriov-network-device-plugin/vendor/golang.org/x/text/unicode/norm
github.com/intel/sriov-network-device-plugin/vendor/golang.org/x/net/internal/timeseries
github.com/intel/sriov-network-device-plugin/vendor/golang.org/x/net/trace
github.com/intel/sriov-network-device-plugin/vendor/golang.org/x/text/secure/bidirule
github.com/intel/sriov-network-device-plugin/vendor/google.golang.org/grpc/grpclog
github.com/intel/sriov-network-device-plugin/vendor/google.golang.org/grpc/connectivity
github.com/intel/sriov-network-device-plugin/vendor/google.golang.org/grpc/credentials
github.com/intel/sriov-network-device-plugin/vendor/golang.org/x/net/idna
github.com/intel/sriov-network-device-plugin/vendor/google.golang.org/grpc/resolver
github.com/intel/sriov-network-device-plugin/vendor/google.golang.org/grpc/codes
github.com/intel/sriov-network-device-plugin/vendor/google.golang.org/grpc/balancer
github.com/intel/sriov-network-device-plugin/vendor/google.golang.org/grpc/internal
github.com/intel/sriov-network-device-plugin/vendor/google.golang.org/grpc/keepalive
github.com/intel/sriov-network-device-plugin/vendor/google.golang.org/grpc/metadata
github.com/intel/sriov-network-device-plugin/vendor/google.golang.org/grpc/naming
github.com/intel/sriov-network-device-plugin/vendor/google.golang.org/grpc/peer
github.com/intel/sriov-network-device-plugin/vendor/google.golang.org/grpc/stats
github.com/intel/sriov-network-device-plugin/vendor/google.golang.org/grpc/tap
github.com/intel/sriov-network-device-plugin/vendor/google.golang.org/grpc/grpclb/grpc_lb_v1/messages
github.com/intel/sriov-network-device-plugin/vendor/github.com/golang/protobuf/ptypes/any
github.com/intel/sriov-network-device-plugin/vendor/golang.org/x/net/lex/httplex
github.com/intel/sriov-network-device-plugin/vendor/github.com/golang/protobuf/ptypes/duration
github.com/intel/sriov-network-device-plugin/vendor/github.com/golang/protobuf/ptypes/timestamp
github.com/intel/sriov-network-device-plugin/vendor/golang.org/x/net/http2
github.com/intel/sriov-network-device-plugin/vendor/google.golang.org/genproto/googleapis/rpc/status
github.com/intel/sriov-network-device-plugin/vendor/github.com/gogo/protobuf/proto
github.com/intel/sriov-network-device-plugin/vendor/github.com/golang/protobuf/ptypes
github.com/intel/sriov-network-device-plugin/vendor/github.com/gogo/protobuf/sortkeys
github.com/intel/sriov-network-device-plugin/vendor/google.golang.org/grpc/status
github.com/intel/sriov-network-device-plugin/vendor/github.com/gogo/protobuf/protoc-gen-gogo/descriptor
github.com/intel/sriov-network-device-plugin/vendor/google.golang.org/grpc/transport
github.com/intel/sriov-network-device-plugin/vendor/github.com/gogo/protobuf/gogoproto
github.com/intel/sriov-network-device-plugin/vendor/google.golang.org/grpc
github.com/intel/sriov-network-device-plugin/vendor/k8s.io/kubernetes/pkg/kubelet/apis/deviceplugin/v1beta1
github.com/intel/sriov-network-device-plugin/cmd/sriovdp
Building sriovdp...
Done!

Reply:
Yes, it looked for glint binary from usual paths. Then it installed it  in ./bin and proceeded with the build:
```
running gofmt...
setting GOPATH...
building golint...
running golint...
```
Subsequent builds should work as normal. If we continue see the same error then we might need to export the $GOBIN as well.
Also can you also please remove existing ./gotpah directory if its still there? We are moving this into ./.gopath and this directory will be updated on each build. This contains a symlink which may point to an absolute path. And when we are creating docker image we are basically copying everything as is into the Docker contexts.